core/commands.inc.sh

```diff
@@ -93,21 +82,27 @@ do_test() {
 # ----------------------------------------------------------------------------
 
 do_install() {
-  do_command install $1
+  local target=$1
+  builder_start_action install:$target || return 0
+  if $MESON_LOW_VERSION; then
+    pushd "$MESON_PATH" > /dev/null
+    ninja install
+    popd > /dev/null
+  else
+    meson install -C "$MESON_PATH"
+  fi
+  builder_finish_action success install:$target
 }
 
 do_uninstall() {
-  do_command uninstall $1
-}
-
-do_command() {
-  local command=$1
-  local target=$2
-  builder_start_action $command:$target || return 0
+  local target=$1
+  builder_start_action uninstall:$target || return 0
   pushd "$MESON_PATH" > /dev/null
-  ninja $command
+  # Note: there is no meson uninstall command, which means
+  # this probably won't work on Windows
```

Comment:
That is correct... if you use the vs backend instead of the ninja backend (not the MSVC compiler, just VS solution files) then you cannot run ninja for this, and that's the only official way to get an uninstall command.

Reply:
Main issue we were having here is that running bare ninja depends on having all the paths just tickety-boo on Windows. So we were trying to use `meson compile` instead of `ninja`, `meson test` instead of `ninja test`, etc. `ninja uninstall` was the only command that didn't have a corresponding `meson uninstall` that we could use.

Reply:
The good news is that `ninja uninstall` doesn't attempt to run the compiler, while `ninja` obviously does -- and `ninja test` less obviously does, because... it tries to rebuild all targets before running the tests. :D

So it is safe to run `ninja uninstall` without having the VS environment set up.